oximeter/db/src/client/mod.rs

```diff
@@ -242,18 +272,8 @@ impl Client {
 
     /// Ping the ClickHouse server to verify connectivitiy.
     pub async fn ping(&self) -> Result<(), Error> {
-        let client = ClientVariant::new(&self.source).await?;
-
-        handle_db_response(
-            client
-                .reqwest()
-                .get(format!("{}/ping", client.url()))
-                .send()
-                .await
-                .map_err(|err| Error::DatabaseUnavailable(err.to_string()))?,
-        )
-        .await?;
-        debug!(self.log, "successful ping of ClickHouse server");
+        self.native_pool.claim().await?.ping().await.map_err(Error::Native)?;
```

Comment:
Just making sure I understand - as of this PR, this is the only use of the native connection pool, right? If something is wrong with the native pool but not the http pool, is it okay that `ping` fails?

Reply:
> Just making sure I understand - as of this PR, this is the only use of the native connection pool, right?

That's right, I'm being pretty conservative :)

> is it okay that ping fails?

The `ping` method is never actually used in production AFAIK, only tests. I actually should probably move it `cfg(test)`. This PR is just adding the pool, and the minimal testing to see if we can use it.